channels/channel.be/vier/chn_vier.py

```diff
@@ -587,27 +588,58 @@ def __update_video(self, item, data):
 
         if ".m3u8" not in m3u8_url:
             Logger.info("Not a direct M3u8 file. Need to log in")
-            url = "https://api.viervijfzes.be/content/%s" % (m3u8_url, )
+            url = "https://api.goplay.be/web/v1/videos/long-form/%s" % (m3u8_url, )
 
             # We need to log in
             if not self.loggedOn:
                 self.log_on()
 
             # add authorization header
             authentication_header = {
-                "authorization": self.__idToken,
+                "authorization": "Bearer {}".format(self.__idToken),
                 "content-type": "application/json"
             }
             data = UriHandler.open(url, additional_headers=authentication_header)
             json_data = JsonHelper(data)
-            m3u8_url = json_data.get_value("video", "S")
+            m3u8_url = json_data.get_value("manifestUrls", "hls")
 
-        # Geo Locked?
-        if "/geo/" in m3u8_url.lower():
-            # set it for the error statistics
-            item.isGeoLocked = True
+        if m3u8_url is not None:
+            if "/geo/" in m3u8_url.lower():
+                # set it for the error statistics
+                item.isGeoLocked = True
 
-        item.complete = M3u8.update_part_with_m3u8_streams(
-            item, m3u8_url, channel=self, encrypted=False)
+            item.complete = M3u8.update_part_with_m3u8_streams(
+                item, m3u8_url, channel=self, encrypted=False)
+
+        else:
+            # Fetch SSAI stream data
+            Logger.info("No stream data found, trying SSAI data")
+            content_source_id = json_data.get_value("ssai", "contentSourceID")
+            video_id = json_data.get_value("ssai", "videoID")
```

Comment:
The `json_data` might not exists, as it is defined in line 603, which is inside an `if`. So the `json_data` is only fetched if ".m3u8" is not in the initial `m3u8_url`. 


I would suggest to wrap the `SSAI` stuff in a method called `def __get_ssai_streams(item, data)` and after line 604, if the `m3u8 is None` check for `content_source_id = json_data.get_value("ssai", "contentSourceID")` and if it is present, `return self.__get_ssai_streams(item, data)` and leave the rest as it was before your changes.

Reply:
Thanks for your feedback.

Part of the change includes some error handling/reporting for locked items, which is returned in the JSON data (without m3u8 URL). I'll keep that in the `__update_video()` method, assuming it'll give more insight in case error messages show up. The previous code would attempt to `lower()` on the `None` m3u8 URL, for the geo-lock check. This raises errors instead of reporting cleanly what was wrong. Do you agree with this approach?